include/dali/core/exec/tasking/task.h

```diff
@@ -355,7 +355,7 @@ class Task : public CompletionEvent {
     } else {
       // A task with a non-scalar return value must return a collection or a tuple.
       // We can check the return type early (i.e. now) to aid debugging.
-      CheckFuncResultType(num_results, std::forward<F>(function));
+      CheckFuncResultType(num_results, function);
```

Comment:
I'm not quite sure about it... It definitely changes what `F` will be in `CheckFuncResultType` and we don't want that. Since F is passed by a universal reference, I don't think std::forward can actually affect the variable `function` - there's no real forwarding happening; note that the value is not used at all in `CheckFuncResultType`.
You could either mark it as intentional or even false positive (there's no real double forwarding) or, as an alternative, do the following:
```suggestion
      CheckFuncResultType<F>(num_results);
```
and then:
```C++
  template <typename F>
  void CheckFuncResultType(int num_results) {
    using Func = std::remove_reference_t<F>;
    if constexpr (std::is_invocable_v<Func, Task *>)
      CheckResultType<decltype(std::declval<F>()(this))>(num_results);
    else
      CheckResultType<decltype(std::declval<F>()())>(num_results);
  }

```